### ISSUE TITLE
Non-Quest NPCs kill cause spam; Blacklisting 'breaks' plugin

### ISSUE THREAD

**Issue by Formula350** [FlyingPikachu/Quests#111]
Thankfully this is not **too** serious of an issue, compared to my other one, I mean it IS but it can be avoided at least!

That being said, here's the pertinent info:
OS: Debian GNU/Linux 7 (wheezy)
----Java(TM) SE Runtime Environment (build 1.7.0_72-b14
----Java HotSpot(TM) 64-bit Server VM (build 24.72-b04, mixed mode)
CraftBukkit: 1.8.0 R0.1 (git-Bukkit-33d5de3)
Quests: Produced in 2.3.4 (build 65), reproduced 2.4.0 (build 75)
Citizens: 2.0.14 (built 1213)   [I believe this is the last version supporting CB 1.8.1]
Sentry: 1.8.4 (build 24)
*(Quick note on why I've not updated to 1.8.3 is simply due to some older, abandoned, but important plugins failing to load the few times I attempted 1.8.3)*

Info about the quest and NPC setup:
There are 13 NPCs, all of which share the same name. 12 of them are very weak and die in 1 hit, but also respawn in 30seconds, purely there as an annoyance/distraction to the player during their combat. Something I noticed during each of the times I created one of these "shadow" versions was when I would give them the Sentry Trait, Essentials would generate a player file and assign them a UUID. Whether or not I can go and delete these Essentials files or not, and have them _stay_ gone, I have yet to test. What I CAN tell you is the issue persists even if that "player file" is deleted. This entire issue could simply come down to the fact that Quests might not be generating a "data file" for these "players" and so it sort of freaks out... Anyways...

Console output:
THIS one is the actual target NPC for the Quest, so I was rather surprised to see it come up when I finally killed it...
05.05 15:40:24 [Server] INFO Quester not found for UUID "2c6f8e22-0cd3-295e-a60c-a330bdb4abce". Consider adding them to the Quester blacklist.

These are all of the "shadow" NPCs. I just ran arround collecting them in attempt to make sure I had them all, so a couple may be duplicates:
```javascript
05.05 15:27:18 [Server] INFO Quester not found for UUID "36963ff0-297a-2c5c-8ffd-77aaf22817fe". Consider adding them to the Quester blacklist.
05.05 15:27:16 [Server] INFO Quester not found for UUID "82204f39-0766-2055-a1e7-04eb368bc518". Consider adding them to the Quester blacklist.
05.05 15:27:16 [Server] INFO Quester not found for UUID "3e01d74e-601c-254a-a70c-7d0d6b00a640". Consider adding them to the Quester blacklist.
05.05 15:27:15 [Server] INFO Quester not found for UUID "9698082e-27b0-2dc4-a986-f2bc7fceabb0". Consider adding them to the Quester blacklist.
05.05 15:27:15 [Server] INFO Quester not found for UUID "e39ceafa-d448-2622-a8b7-c9b3adcd0522". Consider adding them to the Quester blacklist.
05.05 15:27:14 [Server] INFO Quester not found for UUID "f37afcc7-65ae-2ca1-97f7-3c60e060cd28". Consider adding them to the Quester blacklist.
05.05 15:27:14 [Server] INFO Quester not found for UUID "f4049fa1-8911-2887-ad03-16ec51729192". Consider adding them to the Quester blacklist.
05.05 15:27:14 [Server] INFO Quester not found for UUID "76b3e3ff-bdcc-2890-a2a1-4df45c4cfad7". Consider adding them to the Quester blacklist.
05.05 15:27:14 [Server] INFO Quester not found for UUID "c98b6261-129d-26d3-bfce-90e208b29cb6". Consider adding them to the Quester blacklist.
05.05 15:26:22 [Server] INFO Quester not found for UUID "cf0bd877-0032-2680-b553-071b793d8581". Consider adding them to the Quester blacklist.
05.05 15:26:17 [Server] INFO Quester not found for UUID "21c06b50-3c1e-25ab-be22-d3c62bc8a6b0". Consider adding them to the Quester blacklist.
05.05 15:26:10 [Server] INFO Quester not found for UUID "c98b6261-129d-26d3-bfce-90e208b29cb6". Consider adding them to the Quester blacklist.
05.05 15:26:05 [Server] INFO Quester not found for UUID "3e01d74e-601c-254a-a70c-7d0d6b00a640". Consider adding them to the Quester blacklist.
05.05 15:26:01 [Server] INFO Quester not found for UUID "5b999dac-c442-2516-b6df-14b5e1f3b16e". Consider adding them to the Quester blacklist.
05.05 15:25:51 [Server] INFO Quester not found for UUID "f4049fa1-8911-2887-ad03-16ec51729192". Consider adding them to the Quester blacklist.
05.05 15:25:51 [Server] INFO Quester not found for UUID "82204f39-0766-2055-a1e7-04eb368bc518". Consider adding them to the Quester blacklist.
```
Since that clutters up the console something fierce I went about actually adding each of those to the config blacklist, and this is where the issue occurs (which happens to be an issue I previously encountered when blacklisting but thought to be user-error on my part...). If I added **any** of the above UUIDs to the Blacklist, that is when the plugin effectively "breaks". My theory is that Quests is incapable of establishing that this is a true player of **any** sort, and as a result it just blacklists _everyone_, even myself. In hoping it was simply one of the UUIDs or an indentation issue that caused this to occur, so I went about removing them one by one and when that failed I tried indenting just one UUID, all to no avail unfortunately. 

This issue only presented with the update to a UUID blacklist system, working fine using the old Player Name input method. I'm not necessarily proposing to revert back to that method (unless it's possible? but I assume it's at the core of utilizing UUIDs), but perhaps it is possible to add in either a new category to put in the specific NPC IDs or add in flag prefix before the ID number in the current blacklist section which tells Quests to not treat it as a UUID?
Example:
```yaml
quester-blacklist:
-#1417
-#1416
-#338
-21c06b50-3c1e-25ab-be22-d3c62bc8a6b0
```
And just in case it has anything to do with how we've named the NPCs, the ones to which those UUIDs above all belong to are named &7&lErebus, whereas a number of others have a space in their name such as "The Minotaur".

Hopefully this isn't a too big of issue to fix :\ Even if you can't fix it, or anytime soon (as again, it's a bug which CAN be 'turned on and off' and simply not blacklisting any prevents it), I still thank you for all your hard work!
-Form

**Comment by FlyingPikachu** [comment]
Please try the dev build below and let me know if it resolves the issue. Make sure to remove the affected UUIDs from the blacklist before testing the build.

http://ci.ac3-servers.eu/job/Quests/76/

**Comment by Formula350** [comment]
Alright I currently have Build 76 running, and I ran the Purge command on myself thinking it would just delete my quests, not knowing it'd blacklist me too lol In any event, it seems it was helpful as it may have resulted in something of use... When **I** was blacklisted, so was everyone else, even though only my UUID was in the config file...

```yaml
quester-blacklist:
- UUID
- UUID
- UUID
- 9b3e0b53-05fc-4f38-ae21-7c673fd9cd0c
```
I'm almost beginning to wonder, is it something to do with just my server, or is this reproducible? <_>

Reason I wonder is that in addition to the above, the original "quester not found" issue does not seem to be fixed in the latest build. 

Would it be helpful if I ran Quests with debug-mode set to true? If I did, does that produce any ouput that would annoy my players any?

Thanks,
-Form

**Comment by FlyingPikachu** [comment]
I may put a flag on the purge sub-command in the future to say whether to blacklist or not. That behavior is strange, though. Feel free to open a new issue for that.

You did remove the NPC uuids from the blacklist, correct?

That debug setting just outputs additional messages to the console (which may or may not aid in your situation), so it can't hurt.

**Comment by Formula350** [comment]
Hmm, I guess I never even looked at it in the sense that there was two issues I was originally posting about! I just thought that "quester not found" and adding of it's listed UUID into the blacklist which resulted in **everyone** being blacklisted, sorta went hand-in-hand. Now that you mention it though, I suppose they would be individual, so I'll try to separate them into their own issue threads.

And correct there were no NPC **or** player UUIDs present, what I posted in my previous reply is 100% what was in my config file, and is actually how Quests populated it (as I had removed all the '- UUID' lines as well). 

I'll turn on the Debug then and see if it spits out anything additional.

-Form
